src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
+@SuppressWarnings("unchecked")
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int currentCapacity = DEFAULT_CAPACITY;
+    private double threshHold = DEFAULT_CAPACITY * LOAD_FACTOR;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[currentCapacity];
+    }
+
+    private int hash(Object key) {
+        int h;
+        return (key == null) ? 0 : Math.abs((h = key.hashCode()) ^ (h >>> 16));
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = hash(key);
+        int index = hash % currentCapacity;
+        if (buckets[index] == null) {
+            buckets[index] = new Node<>(hash, key, value, null);
+            size++;
+        } else if (buckets[index] != null) {
+            Node<K, V> currentNode;
+            for (currentNode = buckets[index]; currentNode.next != null;
+                    currentNode = currentNode.next) {
+                if (hasSameKeyOrValue(currentNode, key, value)) {
+                    break;
+                }
+            }
+            if (!hasSameKeyOrValue(currentNode, key, value)) {
+                currentNode.next = new Node<>(hash, key, value, null);
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> bucket : buckets) {
```

Comment:
the main strength of hashmap - retrieval with O(1) complexity - your method is of linear complexity. We don't need to iterate through buckets - we can calculate buckets using the key's hash

Reply:
As I understood, there is no need in outer loop. We'd still have to use what was the inner loop for elements within the same bucket. True?